docker-compose.yml

```diff
@@ -0,0 +1,90 @@
+services:
+  postgres:
+    image: postgres:13.7-alpine
+    restart: always
+    environment:
+      POSTGRES_USER: postgres
+      POSTGRES_PASSWORD: zenysis
+      POSTGRES_DB: postgres
+      POSTGRES_HOST_AUTH_METHOD: trust
+    ports:
+      - "5432:5432"
+    volumes:
+      - pgdata-volume:/var/lib/postgresql/data
+    healthcheck:
+      test: [ "CMD-SHELL", "pg_isready -U postgres -d postgres" ]
+      start_period: 0s
+      interval: 10s
+      timeout: 30s
+      retries: 3
+  redis:
+    image: redis
+    ports:
+      - 6379:6379
+    healthcheck:
+      test: [ "CMD", "redis-cli", "--raw", "incr", "ping" ]
+      interval: 10s
+      timeout: 30s
+      start_period: 0s
+      retries: 3
+  hasura:
+    image: hasura/graphql-engine:v2.11.3.cli-migrations-v2
+    ports:
+      - 8088:8080
+    environment:
+      HASURA_GRAPHQL_ENABLE_TELEMETRY: 'false'
+      HASURA_GRAPHQL_MIGRATIONS_SERVER_TIMEOUT: 500
+      HASURA_GRAPHQL_DATABASE_URL: postgresql://postgres:zenysis@postgres:5432/${ZEN_ENV}-local
+    healthcheck:
+      test: "timeout 10s bash -c ':> /dev/tcp/127.0.0.1/8080'"
+      interval: 30s
+      timeout: 30s
+      start_period: 0s
+      retries: 3
+  web:
+    build:
+      dockerfile: Dockerfile_dev
+    volumes:
+      - .:/app
+      - /app/venv
+      - /app/node_modules
+    environment:
+      - ZEN_HOME=/app
+      - PYTHONPATH=/app:/usr/bin
+      - DATABASE_HOST=postgres
+      - DATABASE_USER=postgres
+      - DATABASE_URL=postgresql://postgres:zenysis@postgres:5432/${ZEN_ENV}-local
```

Comment:
I agree - see bigger re-factor notes though. I'll remove here, and rely on defaults in scripts/db/postgres/dev/upgrade_dev_database.sh for now - but really would prefer to remove it entirely. Either only have those, or only have database url!

Reply:
See other note about removing `DATABASE_HOST`, but yeah just `DATABASE_URL` here would be nice.

Is it possible to construct this `DATABASE_URL` using the environment variables from postgres? Not a big deal at all, but if a user wanted to change their local postgres location, they'd have to update two places in the docker compose.

Reply:
There's no clean solution to not duplicating the the DATABASE url.

I could move it to a .env file, so it's only specified once - but seems to defeat the purpose, since the entire reason for having docker-compose is so that the postgress location isn't changing.

There's no way to define a DATABASE_URL within the yaml file. I'd love a better solution - but afraid this is it for now.